### PR TITLE
Refactor Tractor - Add If else code smell with examples

### PR DESCRIPTION
**Pull request by plato721**
Add an if/else code smell with some sample remedies.

### CODE REVIEW

**Comment by plato721** [comment]
@rrgayhart @rmcornell Could you review my refactor tractor PR? Thanks!

File: ruby_04-apis_and_scalability/js_refactor_tractor.md [hunk 1]
```diff
@@ -303,6 +303,141 @@ __Discussion Points__
 
 #### Dead Code Among the Living
 
+#### If If If Else If Else Else If Else ...
+
+Deeply nested if/else statements should be avoided-- they make a program difficult to follow. There are many ways to refactor/restructure to avoid this condition, and good strategy can vary with the situation.
+
+##### 1. Break up into functions.
```
Comment (rrgayhart) [inline ruby_04-apis_and_scalability/js_refactor_tractor.md]: Might want to make these `######`, as they are currently the same header size as the main 'If Else...'

**Comment by rrgayhart** [comment]
Seriously excellent! :ship: it!

File: ruby_04-apis_and_scalability/js_refactor_tractor.md (hunk 1 above)
Comment (plato721) [inline ruby_04-apis_and_scalability/js_refactor_tractor.md]: Looks like #### (4) for topic headers, and #####(5) for the three if/else subtopics? Let me know if it doesn't look right.

Comment (rrgayhart) [inline ruby_04-apis_and_scalability/js_refactor_tractor.md]: I think you're right, must have misread it!

**Comment by plato721** [comment]
Awesome. Somehow I managed to put this in the main repo instead of my fork, so I can't merge it. But here ya go. :-)

**Pull request merged**